libnmstate/schemas/interfaces.json

```diff
@@ -94,5 +101,41 @@
                 "pattern": "^([a-fA-F0-9]{2}:){5}[a-fA-F0-9]{2}$"
             }
         }
+    },
+    "definitions-ethernet": {
+        "interface": {
+            "allOf": [
+                {
+                    "properties": {
+                        "type": { "type": "string", "enum": ["ethernet"] }
+                    }
+                },
+                {
+                    "oneOf": [
+                        { "$ref": "#/__definitions__/auto-false" },
+                        { "$ref": "#/__definitions__/auto-true" }
+                    ]
+                }
+            ],
+            "required": [ "type", "auto-negotiation" ]
```

Comment:
i don't like the fact that we need to require something that is hidden in reference. is there a way around? if not, please keep one requirement per line and add inline comment to where is it required from.